pkg/api/auth_middleware.go

```diff
@@ -148,11 +160,107 @@ func enhanceWithFriendlyName(user *model.User, friendlyName string) *model.User
 	return user
 }
 
+// userFromSAML returns a user from an existing SAML session.
+// If the user doesn't exist on the lakeFS side, it is created.
+// This function does not make any calls to an external provider.
+func userFromSAML(ctx context.Context, logger logging.Logger, authService auth.Service, authSession *sessions.Session, cookieAuthConfig *config.CookieAuthVerification) (*model.User, error) {
+	idTokenClaims, ok := authSession.Values[SAMLTokenClaimsSessionKey].(oidc_encoding.Claims)
+
+	if idTokenClaims == nil {
+		return nil, nil
+	}
+
+	if !ok {
+		logger.WithField("claims", authSession.Values[SAMLTokenClaimsSessionKey]).Error("failed decoding tokens")
+		return nil, ErrAuthenticatingRequest
```

Comment:
You can wrap error to gain more information:
```suggestion
		return nil, fmt.Errorf("getting token claims: %w", ErrAuthenticatingRequest)
```

Reply:
thanks, done